src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,61 @@
+body {
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  width: 100vw;
+  height: 100vh;
+}
+
+  .calendar {
+    flex-wrap: wrap;
+    gap: $calendar-gap;
+    width: calc(7 * $calendar-day-width + 7px);
+    list-style: none;
+    display: flex;
+
+    &__day {
+      cursor: pointer;
+      background-color: $calendar-day-color;
+      border: 1px solid black;
+      justify-content: center;
+      align-items: center;
+      height: $calendar-day-height;
+      width: $calendar-day-width;
+      transition: 500ms;
+      display: flex;
+      box-sizing: border-box;
+
+      &:hover {
+        background-color: #FFBFCB;
+        transform: translateY(-20px);
+
+      }
+
+      &::before {
+        font-family: Arial, Helvetica, sans-serif;
+        font-size: $calendar-day-font-size;
+        content: "";
+      }
+
+      @for $number from 1 through $calendar-length {
+        &:nth-child(#{$number})::before {
+          content: "#{$number}";
+        }
+      }
+    }
+
+    @each $day, $index in $days {
+      &--start-#{$day} > :first-child {
+        margin-left: ($index - 1) * ($calendar-day-width + $calendar-gap) ;
+      }
+    }
+
+    @for $i from $calendar-length through 31 {
+      &__day:nth-child(#{$i + 1}) {
+        display: none;
+      }
```

Comment:
Ensure that the `$calendar-length` variable accurately represents the number of days in the month. The loop should hide the days that are not part of the current month. If `$calendar-length` is set dynamically, this loop will correctly hide the excess days, otherwise, you might need to adjust the logic.